modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/disruptor/inbound/InboundKernelOpsEvent.java

```diff
@@ -127,7 +127,9 @@ public void startMessageDelivery() {
      * Stop message delivery threads in Andes
      */
     public void stopMessageDelivery() {
-        messagingEngine.stopMessageDelivery();
+        if(messagingEngine != null){
```

Comment:
Since we are trying to shut down the carbon kernel, messagingEngine is not initialized. But, shutdown method is a general one for all scenarios.